### PR TITLE
Code cleanup 

### PR DESCRIPTION
**Pull request by loxal**
In order to reduce compiler warnings and code smells.

* more consistent usage of append 
* comparing objects w/o ==
* public modifier is superfluous in interface definitions, also no abstract methods there
* fix erroneous overriding (potential runtime issues) of equals (if an obj is an instance of something, it cannot be null)
* fix clone signature
* make pseudo-constants actual constants
* private methods cannot be overridden, hence making them final is just noisy

### CODE REVIEW

File: ethereumj-core/src/main/java/org/ethereum/core/BlockchainImpl.java [hunk 1]
```diff
@@ -777,7 +790,7 @@ private void addReward(Block block) {
         if (block.getUncleList().size() > 0) {
             for (BlockHeader uncle : block.getUncleList()) {
                 track.addBalance(uncle.getCoinbase(),
-                        new BigDecimal(BLOCK_REWARD).multiply(BigDecimal.valueOf(8 + uncle.getNumber() - block.getNumber()).divide(new BigDecimal(8))).toBigInteger());
+                        new BigDecimal(BLOCK_REWARD).multiply(BigDecimal.valueOf(8 + uncle.getNumber() - block.getNumber()).divide(new BigDecimal(8), RoundingMode.HALF_UP)).toBigInteger());
```
Comment (Nashatyrev) [inline ethereumj-core/src/main/java/org/ethereum/core/BlockchainImpl.java]: Alex, can you please change this to `BigInteger` calculation. There is no place for decimal. 
Thanks we always had even `BLOCK_REWARD`, else we definitely get a conflict here.

**Comment by loxal** [comment]
@Nashatyrev the build fails because of a failing test.
> org.ethereum.jsontestsuite.GitHubBlockTest#runBCRPCAPITest
I guess the test assumes Git commits that are not present in my forked repository.

File: ethereumj-core/src/main/java/org/ethereum/datasource/redis/RedisSet.java [hunk 2]
```diff
@@ -39,6 +38,7 @@ public boolean contains(final Object o) {
         return pooledWithResult(new Function<Jedis, Boolean>() {
             @Override
             public Boolean apply(Jedis jedis) {
+                if(o instanceof Boolean) System.out.println("");
```
Comment (Nashatyrev) [inline ethereumj-core/src/main/java/org/ethereum/datasource/redis/RedisSet.java]: This is mostly deprecated class but anyways... :)

File: ethereumj-core/src/main/java/org/ethereum/core/BlockchainImpl.java (hunk 1 above)
Comment (loxal) [inline ethereumj-core/src/main/java/org/ethereum/core/BlockchainImpl.java]: My fault, apparently the rounding strategy came from an unmerged (and broken) PR. I’ve also changed BigDecimal to BigInteger.

Comment (loxal) [inline ethereumj-core/src/main/java/org/ethereum/core/BlockchainImpl.java]: Assuming the int-based calculation is implemented as desired, we need to adjust the tests.

Comment (Nashatyrev) [inline ethereumj-core/src/main/java/org/ethereum/core/BlockchainImpl.java]: What tests need to be adjusted?

Comment (loxal) [inline ethereumj-core/src/main/java/org/ethereum/core/BlockchainImpl.java]: The CI job fails because of the seitch from BigDecimal to BigInteger.

Comment (Nashatyrev) [inline ethereumj-core/src/main/java/org/ethereum/core/BlockchainImpl.java]:
```
BLOCK_REWARD.multiply(BigInteger.valueOf(8 + uncle.getNumber() - block.getNumber())).divide(BigInteger.valueOf(8))
```

Comment (Nashatyrev) [inline ethereumj-core/src/main/java/org/ethereum/core/BlockchainImpl.java]: If JSON test (TCK in fact) fails then our implementation need to be adjusted most likely :)

**Comment by loxal** [comment]
@Nashatyrev ok, I fixed the broken build incl. the conversion to BigInteger that you have proposed.

**Comment by Nashatyrev** [comment]
Cool, thanks Alex!

**Pull request merged**